### PR TITLE
Hadas and Sergey prerelease support final!

### PR DESCRIPTION
**Pull request by hadasfa**
Create prerelease version for every commit with [prerelease]

### CODE REVIEW

File: .github/workflows/prerelease.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    if: "contains(github.event.head_commit.message, '[prerelease]')"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: ${{ github.head_ref }}
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 16
```
Comment (orrgottlieb) [inline .github/workflows/prerelease.yml:18]: please change to ` node-version-file: ".nvmrc"` so we will always be aligned

File: .github/workflows/prerelease.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        if: false
+        id: wait-for-tests
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          checkName: "test"
+          ref: ${{ github.event.pull_request.head.sha || github.sha }}
+          timeoutSeconds: 1200
+          intervalSeconds: 20
+      - name: "Tests failed - exit"
+        if: false && steps.wait-for-tests.outputs.conclusion != 'success'
```
Comment (orrgottlieb) [inline .github/workflows/prerelease.yml:37]: should this be `false` or this is the tests?

**Review by orrgottlieb — APPROVED** [review]

File: .github/workflows/prerelease.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        uses: 8BitJonny/gh-get-current-pr@2.2.0
+        id: PR
+        if: success()
+      - name: Create comment with prerelease version number
+        if: success()
+        uses: peter-evans/create-or-update-comment@v1
+        with:
+          issue-number: ${{ steps.PR.outputs.number }}
+          body: |
+            Prerelease version published: `${{ steps.create-update-version.outputs.package_version }}`
```
Comment (orrgottlieb) [inline .github/workflows/prerelease.yml:72]: for DevEx - we could add like
```bash
npm i monday-ui-react-core@###version###

yarn add monday-ui-react-core@###version###

```

File: .github/workflows/prerelease.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        env:
+          BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+          COMMIT_SHA: ${{ github.event.pull_request.head.sha || github.sha }}
+        run: |
+          npm run prerelease
+          echo "package_version=$(node -e "console.log(require('./package.json').version)")" >> $GITHUB_OUTPUT
+      - name: Publish the new version to npm
+        if: success()
+        run: | # Compare the latest version and the current version and publish only if the version changed
+          if [ $(npm view monday-ui-react-core version) != ${{ steps.create-update-version.outputs.package_version }} ]; then npm run build && npm publish --tag prerelease; else echo ''; fi;
```
Comment (sahariko) [inline .github/workflows/prerelease.yml:53]: Wouldn't `npm view monday-ui-react-core version` always be different to `steps.create-update-version.outputs.package_version`?

Comment (sahariko) [reply .github/workflows/prerelease.yml:53]: Also, this can be in the `if` key instead.

File: .github/workflows/prerelease.yml (hunk 2 above)
Comment (hadasfa) [reply .github/workflows/prerelease.yml:37]: for tests, will be removed

Comment (orrgottlieb) [reply .github/workflows/prerelease.yml]: thanks

File: .github/workflows/prerelease.yml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    if: "contains(github.event.head_commit.message, '[prerelease]')"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: ${{ github.head_ref }}
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ".nvmrc"
```
Comment (orrgottlieb) [inline .github/workflows/prerelease.yml:18]: `node-version-file: ".nvmrc"`

**Pull request merged**